File: 56-MergeIntervals/56-MergeIntervals.py
# Last updated: 7/12/2026, 6:24:28 PM
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if not intervals: return []

        intervals.sort(key = lambda x : x[0])

        merged = intervals[0:1]

        for i in range(1, len(intervals)):
            if merged[-1][1] >= intervals[i][0]:
                merged[-1][0] = min(merged[-1][0], intervals[i][0])
                merged[-1][1] = max(merged[-1][1], intervals[i][1])
            else:
                merged.append(intervals[i])

        return merged




        

File: 56-MergeIntervals/56-MergeIntervals.java
// Last updated: 7/12/2026, 6:24:36 PM
class Solution {
    public int[][] merge(int[][] intervals) {

        Arrays.sort(intervals,(i1,i2) -> Integer.compare(i1[0],i2[0]));

        ArrayList<int[]> output = new ArrayList<>();

        int start = intervals[0][0];
        int end = intervals[0][1];

        for(int i = 1; i < intervals.length; i++){
            int s = intervals[i][0];
            int e = intervals[i][1];

            if(s <= end){
                end = Math.max(end,e);
            }else{
                output.add(new int[] {start,end});
                start = s;
                end = e;
            }
        }

        output.add(new int[]{start,end});

         return output.toArray(new int[output.size()][2]); 
        
    }
}